src/new/connection.ts

```diff
@@ -66,7 +66,10 @@ class WalletConnection extends BaseConnection {
       connectionTimeout?: number;
       wsURL: string;
       logger: ILogger;
-    } = { wsURL: helpers.getWSServerURL(this.currentServer), logger: options.logger };
+    } = {
+      wsURL: helpers.getWSServerURL(this.currentServer),
+      logger: options.logger || getDefaultLogger(),
+    };
```

Comment:
_Unrelated to PR objective_
This is an error found during testing.
The default logger from the websocket class does not work when passing undefined logger here.
So if there is no logger, we should use the default.